### PR TITLE
Remove the `rustc-serialize` crate and use the `hex` crate instead.

### PR DESCRIPTION
**Pull request by jeandudey**

### CODE REVIEW

**Comment by svyatonik** [comment]
There are some compilation errors when tests are running (`cargo test --all`). Could you please fix?

**Comment by jeandudey** [comment]
@svyatonik done :+1:

**Comment by jeandudey** [comment]
@svyatonik Thanks for the review, I addressed some of your changes and removed `hex` in favor of `rustc-hex`, however using `hex-literal` for tests seems to be an overkill both for maintenance and compile times, and also adds some confusion (two crates to parse hexadecimal strings?).

File: chain/src/transaction.rs [hunk 1]
```diff
@@ -100,7 +100,8 @@ pub struct Transaction {
 
 impl From<&'static str> for Transaction {
 	fn from(s: &'static str) -> Self {
-		deserialize(&s.from_hex().unwrap() as &[u8]).unwrap()
+        let vec: Vec<u8> = s.from_hex().unwrap();
```
Comment (svyatonik) [inline chain/src/transaction.rs]: Please use tabs

File: primitives/src/bytes.rs [hunk 2]
```diff
@@ -84,7 +84,8 @@ impl io::Write for Bytes {
 
 impl fmt::Debug for Bytes {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-		f.write_str(&self.0.to_hex())
+        let s: String = self.0.to_hex();
```
Comment (svyatonik) [inline primitives/src/bytes.rs]: Please use tabs

File: primitives/src/hash.rs [hunk 3]
```diff
@@ -81,13 +81,15 @@ macro_rules! impl_hash {
 
 		impl fmt::Debug for $name {
 			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				f.write_str(&self.0.to_hex())
+                let s: String = self.0.to_hex();
```
Comment (svyatonik) [inline primitives/src/hash.rs]: Please use tabs

File: primitives/src/hash.rs [hunk 4]
```diff
@@ -81,13 +81,15 @@ macro_rules! impl_hash {
...
+				f.write_str(&s)
 			}
 		}
 
 		impl fmt::Display for $name {
 			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				f.write_str(&self.0.to_hex())
+                let s: String = self.0.to_hex();
```
Comment (svyatonik) [inline primitives/src/hash.rs]: Please use tabs

File: test-data/src/block.rs [hunk 5]
```diff
@@ -312,6 +312,7 @@ impl<F> TransactionBuilder<F> where F: Invoke<chain::Transaction> {
 			//0xfd...(0xffff + 3) => 3,
 			0xfd...0x10002 => 3,
 			//0x10000...(0xffff_ffff + 5) => 5,
+            #[cfg(target_pointer_width = "64")]
```
Comment (svyatonik) [inline test-data/src/block.rs]: Is that intentional?

Comment (jeandudey) [reply test-data/src/block.rs]: Ooops no, let me delete it, it was a one-time hack to build on i686. That constant overflows `usize`, that's content for another PR.

**Pull request merged**